driver-core/src/main/com/mongodb/internal/connection/AwsAuthenticator.java

```diff
@@ -194,13 +173,37 @@ public void dispose() {
             BsonDocument ret = new BsonDocument()
                     .append("a", new BsonString(authorizationHeader.toString()))
                     .append("d", new BsonString(authorizationHeader.getTimestamp()));
-            if (token != null) {
-                ret.append("t", new BsonString(token));
+            if (sessionToken != null) {
+                ret.append("t", new BsonString(sessionToken));
             }
 
             return toBson(ret);
         }
 
+        private AwsCredential createAwsCredential() {
```

Comment:
Centralized the logic for creating the AwsCredential.  This code can't be shared with CSFLE, but I think that's ok since the first two branches are going to be slightly different in that context anyway.